### DISCUSSION TITLE
Output non generic errors

### DISCUSSION THREAD

**Discussion by kotyara85** [oapi-codegen/oapi-codegen#567] [Q&A]
Hello,
I noticed, oapi generated files would return some generic errors, for example auth is just this
`Security requirements failed`

Is there a way to return an actual error?
Thanks

**Comment by jamietanna** [comment]
This is actually [returned from an upstream library](https://github.com/getkin/kin-openapi/blob/cd4e291bcf5b739503e9c5e169d08c628b58a596/openapi3filter/errors.go#L80-L82) so it may be worth raising a feature request there.

Alternatively, we could add some additional code to wrap the error - what sort of additional information would you expect?

**Reply by kotyara85** [reply]
Just pass an error would satisfy, if it’s nil a standard error can be returned. Thanks!

**Reply by jamietanna** [reply]
An error handler can be configured when preparing the filter ([code](https://github.com/deepmap/oapi-codegen/blob/9a9e66fe11c82b21732777233127d518ad092481/pkg/chi-middleware/oapi_validate.go#L28)) so you can do something more with it.

https://github.com/deepmap/oapi-codegen/issues/639 may be related if you want the raw error, but the raw error doesn't have any other information, so would need an upstream change
